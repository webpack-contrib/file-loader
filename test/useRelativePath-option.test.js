import webpack from './helpers/compiler';

describe('when applied with `useRelativePath` option', () => {
  it('matches snapshot for `false` value (`{Boolean}`)', async () => {
    const config = {
      loader: {
        test: /(png|jpg|svg)/,
        options: {
          useRelativePath: false,
        },
      },
    };

    const stats = await webpack('nested/fixture.js', config);
    const [module] = stats.toJson().modules;
    const { assets, source } = module;

    expect({ assets, source }).toMatchSnapshot();
  });

  it('matches snapshot for `true` value (`{Boolean}`)', async () => {
    const config = {
      loader: {
        test: /(png|jpg|svg)/,
        options: {
          useRelativePath: true,
        },
      },
    };

    const stats = await webpack('nested/fixture.js', config);
    const [module] = stats.toJson().modules;
    const { assets, source } = module;

    expect({ assets, source }).toMatchSnapshot();
  });

  it('matches snapshot for `true` value (`{Boolean}`) and with relative `context`', async () => {
    const config = {
      loader: {
        test: /(png|jpg|svg)/,
        options: {
          context: './test/fixtures/nested/',
          useRelativePath: true,
        },
      },
    };

    const stats = await webpack('nested/fixture.js', config);
    const [module] = stats.toJson().modules;
    const { assets, source } = module;

    expect({ assets, source }).toMatchSnapshot();
  });

  it('matches snapshot for `true` value (`{Boolean}`) and with absolute `context`', async () => {
    const config = {
      loader: {
        test: /(png|jpg|svg)/,
        options: {
          context: '../nested/',
          useRelativePath: true,
        },
      },
    };

    const stats = await webpack('nested/fixture.js', config);
    const [module] = stats.toJson().modules;
    let { assets, source } = module;

    if (process.env.CIRCLECI) {
      assets = [assets[0].replace('project', 'file-loader')];
      source = source.replace('project', 'file-loader');
    }

    expect({ assets, source }).toMatchSnapshot();
  });
});
