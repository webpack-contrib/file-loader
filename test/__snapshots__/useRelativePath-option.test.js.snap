// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`when applied with \`useRelativePath\` option matches snapshot for \`false\` value (\`{Boolean}\`) 1`] = `
Object {
  "assets": Array [
    "9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = __webpack_public_path__ + \\"9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;

exports[`when applied with \`useRelativePath\` option matches snapshot for \`true\` value (\`{Boolean}\`) 1`] = `
Object {
  "assets": Array [
    "assets/9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = __webpack_public_path__ + \\"assets/9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;

exports[`when applied with \`useRelativePath\` option matches snapshot for \`true\` value (\`{Boolean}\`) and with absolute \`context\` 1`] = `
Object {
  "assets": Array [
    "file-loader/test/fixtures/nested/assets/9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = __webpack_public_path__ + \\"file-loader/test/fixtures/nested/assets/9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;

exports[`when applied with \`useRelativePath\` option matches snapshot for \`true\` value (\`{Boolean}\`) and with relative \`context\` 1`] = `
Object {
  "assets": Array [
    "assets/9c87cbf3ba33126ffd25ae7f2f6bbafb.png",
  ],
  "source": "module.exports = __webpack_public_path__ + \\"assets/9c87cbf3ba33126ffd25ae7f2f6bbafb.png\\";",
}
`;
